weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallChat.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * Code to determine if a call is a chat call, and if so,
+ * load the message data and render it.
+ */
+
+import _ from 'lodash';
+import React, {useEffect, useMemo, useState} from 'react';
+
+import {LoadingDots} from '../../../../../LoadingDots';
+import {isWeaveRef} from '../../filters/common';
+import {MessageList} from '../PromptPage/MessageList';
+import {useWFHooks} from '../wfReactInterface/context';
+import {
+  CallSchema,
+  Loadable,
+} from '../wfReactInterface/wfDataModelHooksInterface';
+
+type CallChatProps = {call: CallSchema};
+
+// Does this look like a message object?
+export const isMessage = (message: any): boolean => {
+  if (!_.isPlainObject(message)) {
+    return false;
+  }
+  // TODO: Check for role: string?
+  if (!('content' in message) && !('tool_calls' in message)) {
+    return false;
+  }
+  return true;
+};
+
+// Does this call look like a chat formatted object?
+export const isCallChat = (call: CallSchema): boolean => {
+  if (!('messages' in call.rawSpan.inputs)) {
+    return false;
+  }
+  const {messages} = call.rawSpan.inputs;
```

Comment:
we should be avoiding `rawSpan` as much as possible and instead using the `traceCall` (TraceCallSchema)

Reply:
```
  // See note above `RawSpanFromStreamTableEra` regarding the `rawSpan` field.
  // We should wean off of this and move the needed properties to `CallSchema`.
  rawSpan: RawSpanFromStreamTableEra;
```